### DISCUSSION TITLE
Comparing String and Number in Rego should evaluate to error

### DISCUSSION THREAD

**Discussion by SinghCoder** [open-policy-agent/community#50] [OPA and Rego] [answered]
In the styra's opa-rego course's Quiz on iteration for arrays, objects, and sets, there was this one question:

```js
# What does 'foo' evaluate to? 

n := {"a": 7, "b": 1, "c": 3, "d": 9} 
foo { 
   some i 
      v := n[i] 
      i > 7 
}
```

and in the answer it is given that:

```
Keys in objects must be strings, greater-than evaluates to true when comparing a string and a number, which means that this rule will succeed, and foo will evaluate to true.
```

I did not get that, why `>` evaluates to true when comparing string and number, those two things technically are not comparable, so shouldn't it return an error instead?

Similar thing holds when we compare a boolean with a number.

An example in playground: https://play.openpolicyagent.org/p/FAtMFxUVTZ

**Comment by anderseknert** [comment, marked as answer]
I agree that this is slightly confusing, and if designed today this could probably have been made an error. Considering how uncommon these types of comparisons likely are, and that changing this now would break existing policies (however uncommon) I don't believe it would be desirable to have this changed now. Good to have this reported here though if others should search for it 👍
